tests/ui/split_with_space.rs

```diff
@@ -0,0 +1,16 @@
+//@aux-build:proc_macros.rs
+//@no-rustfix
+
+#[deny(clippy::split_with_space)]
+#[allow(unused)]
+fn main() {
+    let some_space_delimtetered_string = "Hello everyone! How are you doing?";
+
+    for substr in some_space_delimtetered_string.split(' ') {
+        println!("{substr}");
+    }
+
+    for substr in some_space_delimtetered_string.split(" ") {
+        println!("{substr}");
+    }
```

Comment:
There should be some negative tests, e.g. splitting by something that isn't a space.